### ISSUE TITLE
x-end-motor/idler centres different from Z-motor mount - URGENT!

### ISSUE THREAD

**Issue by droftarts** [josefprusa/PrusaMendel#38]
I have carefully measured this in Netfabb, and noticed it when I printed out the felt branch. Distance of motor centre (or Z-leadscrew centre) to Z-rod centre on Z-motor-mount = 30mm, same distance on X-end-motor and X-end-idler = 29mm. Which means the Z-axis motor coupler will ALWAYS run eccentrically.

And this is made even worse as NEMA17 motors are 42.3mm square on the end http://www.zappautomation.co.uk/pdf/steppermotors/SY42STH47-1684A.pdf , so 21.15mm from centre to edge, while the distance from the centre of the motor shaft to the 'wall' between the motor and the Z-rod is 20.8mm, so motor is pushed EVEN FURTHER away from the Z-rod. I guess this is why triffid in Issue 20 https://github.com/prusajr/PrusaMendel/pull/20 moved the motor centre, but in doing so causes the X-leadscrew to run EVEN MORE eccentrically as the motor centre moves to 31mm from Z-rod centre.

I suggest removing 1mm of the wall from between the motor and Z-rod, to allow NEMA17 motors to fit and maintain the 30mm spacing. The X-end-motor and X-end-idler need to be amended to have the 30mm spacing between Z-leadscrew centre and Z-rod centre, which is currently 29mm. I think this may have been an issue for a long time!

**Comment by josefprusa** [comment]
both x-ends use same core from x-end.scad so there is no way it can have different spacing

**Comment by droftarts** [comment]
Yes, both x-ends (x-end-motor and x-end-idler) have the same spacing; 29mm. But this is different from the Z-motor-mount; 30mm. And NEMA17 motors (at least the ones I have) do not fit centrally in the Z-motor-mount, which increases the spacing to 30.35 at best, possibly more.

**Comment by droftarts** [comment]
I have had another close look at the stl files, and I was wrong, the distance between Z-rod and Z-leadscrew centre on the x-ends is 29.5mm, but this is still different from the Z-motor-mount at 30mm, and that the Z-motor-mount does not accommodate (at least my) NEMA17 motors correctly. I have also looked at the scad files, and have found the problem in the bushing.scad file. The centres for the Z-axis smooth rod are set 9.5mm from the origin in both the PLA and LM8UU bushing modules. In the x-end, the Z-axis leadscrew is -20mm from the origin; 20+9.5 = 29.5mm.

This is probably why no one can tighten up the motors on the Z-mount. A stiff coupling, along with the Z-leadscrew running eccentrically, will cause the motor to stall. I can see on my Prusa that the Z-rod and Z-leadscrew are not parallel on both sides.

I think my initial suggestion for changes is still the best course of action.

**Comment by thehans** [comment]
I've also been trying to deal with this issue.  I have measured the STLs in freecad independently and came up with the same measurements as droftarts.  30mm is the spacing on the Z motor mount, and 29.5mm is the spacing on the x-end motor.  I did not measure x-end idler, but i expect it is the same.

My Z motor mounts also will not fit my motor in the center, due to the width of the motor housing.  NEMA 17 is for 1.7", which translates to 43.18mm.  There is just not enough room for the housing to fit, it bumps again the side where the rod clamp nut holes are.  

In addition, if anyone uses nylock m3 nuts (me), these are 4mm tall, and the nut hole depth is only 3mm, forcing the motor out of center an extra 1mm.

**Comment by droftarts** [comment]
An additional problem with the current z-motor-mounts, which have material removed where the M8 studding runs through it, is that if you do the M8 nuts up tight, the 'wall' between the motor and the z-smooth-rod can bend. This further increases the spacing. Either some additional bracing material, like the other end, or filling in the area solidly, like the old version of the mount, is the best solution. Or don't tighten up the nuts too much.
